src/llmcompressor/modifiers/awq/base.py

```diff
@@ -0,0 +1,746 @@
+    :param group_size: number of weights to group together for scaling
+    :param max_chunk_memory: maximum memory to use for each chunk of input activations
+    :param bits: number of bits to quantize the weights to
+    :param symmetric: whether to use symmetric quantization
+    :param duo_scaling: whether to use duo scaling, which uses both input activations
+        and weights to determine the scaling factor
+    :param apply_clip: whether to apply clipping to the weights after scaling
+    """
+
+    # Allow arbitrary types because AWQMapping has fields of type torch.nn.Module
+    model_config: ConfigDict = ConfigDict(arbitrary_types_allowed=True)
+
+    mappings: List[AWQMapping] = DEFAULT_AWQ_MAPPINGS
+    ignore: Optional[List[str]] = None
+    num_calibration_steps: Optional[int] = None
+    calibration_function: Optional[Callable] = None
+    group_size: int = 128
+    max_chunk_memory: int = 1024 * 1024 * 1024
+    bits: int = 4
+    symmetric: bool = True
+    duo_scaling: bool = True
+    apply_clip: bool = True
+
+    resolved_mappings_: Optional[List[ResolvedMapping]] = None
+    scales_: Optional[Dict] = None
+    module_kwargs_: Optional[Dict] = None
+
+    def on_initialize(self, state: State, **kwargs) -> bool:
+        """
+        Initialize and run AWQ on the given state
+
+        :param state: state to run AWQ on
+        :return: True on a successful run, False otherwise
+        """
+        if not (self.end is None or self.end == -1):
+            raise ValueError(
+                f"{self.__class__.__name__} can only be applied during one-shot. "
+                f" Expected end to be None or -1, got {self.end}"
+            )
+        if self.start and self.start != -1:
+            raise ValueError(
+                f"{self.__class__.__name__} can only be applied during one-shot. "
+                f"Expected start to be None or -1, got {self.end}"
+            )
+
+        self.ignore = [] if not self.ignore else self.ignore
+        self.resolved_mappings_ = self._get_resolved_mappings(state.model)
+        self.scales_ = {}
+
+        calibration_dataloader = state.data.calib
+
+        with calibration_forward_context(state.model):
+            self._set_module_kwargs(state.model, calibration_dataloader)
+            self._setup_scale_hooks()
+            self._calibrate(state.model, calibration_dataloader)
+            self._concat_collected_activations()
+            self._apply_smoothing(state.model)
+
+        return True
+
+    def on_finalize(self, state: State, **kwargs) -> bool:
+        """
+        Clean up by clearing the scale and mapping data
+
+        :param state: unused
+        :return: True
+        """
+        if self.scales_ is not None:
+            self.scales_.clear()
+        if self.resolved_mappings_ is not None:
+            self.resolved_mappings_.clear()
+
+        return True
+
+    def _get_resolved_mappings(self, model: Module) -> List[ResolvedMapping]:
```

Comment:
This is the same exact function used by smoothquant, consider moving this to a shared namespace

Reply:
While both modifiers functionally convert a mapping config to the resolved mappings with pointers to torch modules, the logic is not identical. Maybe we can chat in 1:1 if this is worthwhile to try to merge them into a single implementation?